com.unity.render-pipelines.core/Editor/Lighting/GUIFramework/ClickAction.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using UnityEngine;
+
+namespace UnityEditor
+{
+    /// <summary>
+    /// Action for click
+    /// </summary>
+    public class ClickAction : HoveredControlAction
+    {
+        private int m_Button;
+        private bool m_UseEvent;
+        /// <summary>
+        /// Click count
+        /// </summary>
+        public int clickCount = 1;
+        /// <summary>
+        /// Action during onClick
+        /// </summary>
+        public Action<IGUIState, Control> onClick;
+        private int m_ClickCounter = 0;
+
+        /// <summary>
+        /// Initializes and returns an instance of ClickAction
+        /// </summary>
+        /// <param name="control">Current control</param>
+        /// <param name="button">Button ID</param>
+        /// <param name="useEvent">If use an event</param>
+        public ClickAction(Control control, int button, bool useEvent = true) : base(control)
+        {
+            m_Button = button;
+            m_UseEvent = useEvent;
+        }
+
+        /// <summary>
+        /// Get if the trigger condition is validated or not
+        /// </summary>
+        /// <param name="guiState">The gui state</param>
+        /// <returns></returns>
+        protected override bool GetTriggerCondition(IGUIState guiState)
+        {
+            if (guiState.mouseButton == m_Button && guiState.eventType == EventType.MouseDown)
+            {
+                if (guiState.clickCount == 1)
+                    m_ClickCounter = 0;
+
+                ++m_ClickCounter;
+
+                if (m_ClickCounter == clickCount)
+                    return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Calls the methods in its invocation list when triggered
+        /// </summary>
+        /// <param name="guiState">The gui state</param>
+        protected override void OnTrigger(IGUIState guiState)
+        {
+            base.OnTrigger(guiState);
+
+            if (onClick != null)
+                onClick(guiState, hoveredControl);
+
+            if (m_UseEvent)
+                guiState.UseEvent();
+        }
+
+        /// <summary>
+        /// Get Finish Condition
```

Comment:
Checks to see if the finish condition has been met or not. For a ClickAction, this is always true.